src/components/date_picker/super_date_picker/super_date_picker.test.tsx

```diff
@@ -167,6 +167,15 @@ describe('EuiSuperDatePicker', () => {
       );
     });
 
+    test('should invoke onFocus on SuperDatePicker', () => {
+      const focusMock = jest.fn();
+      const componentFocus = shallow<EuiSuperDatePickerInternal>(
+        <EuiSuperDatePicker onTimeChange={noop} onFocus={focusMock} />
+      );
+      componentFocus.dive().simulate('focus');
+      expect(focusMock).toBeCalledTimes(1);
+    });
```

Comment:
This test doesn't actually confirm that focusing on the inner components triggers the on focus - we need to dive into the component to find the buttons that we've passed `onFocus` to.

```suggestion
    it('invokes onFocus callbacks on the date popover buttons', () => {
      const focusMock = jest.fn();
      const component = mount(
        <EuiSuperDatePicker onTimeChange={noop} showUpdateButton={false} />
      );

      component.find('button[data-test-subj="superDatePickerShowDatesButton"]').simulate('click');
      expect(focusMock).toBeCalledTimes(1);

      component.update(); // not 100% sure if this is needed, including just in case

      component.find('button[data-test-subj="superDatePickerstartDatePopoverButton"]').simulate('focus');
      expect(focusMock).toBeCalledTimes(2);

      component.find('button[data-test-subj="superDatePickerstartDatePopoverButton"]').simulate('focus');
      expect(focusMock).toBeCalledTimes(3);
    });
```